### DISCUSSION TITLE
Allow resolving props asynchronously

### DISCUSSION THREAD

**Discussion by nhedger** [vuejs/router#2325] [Ideas]
It would be nice if the props of a component could be resolved asynchronously.

```ts
{
    path: "/",
    name: "home",
    component: HomeView,
    props: async (route) => {
        return {
            posts: await ofetch("https://jsonplaceholder.typicode.com/posts"),
        };
    },
},
```

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/vue-router/issues/1923

**Reply by nhedger** [reply]
Ah, thanks ! Must have missed it. I'll try the meta workaround.

**Reply by posva** [reply]
Nowadays, there is an upcoming data fetching solution that fits the use case even better: https://uvr.esm.is/data-loaders/
